### PR TITLE
Update SUT

### PR DESCRIPTION
**Pull request by kishkaru**

### CODE REVIEW

File: lib/credentials.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#++
+
+module SUT
+  class Credentials
+    def self.get_credentials_simple(session, email)
+      session.execute_async("SELECT * FROM killrvideo.user_credentials WHERE email='#{email}'")
```
Comment (stamhankar999) [inline lib/credentials.rb]: Consider using a prepared-statement here.

File: lib/metrics.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+#++
+
+module SUT
+  class Metrics
+    class Statistic
+      attr_accessor :throughput, :latency, :num_errors, :errors
+
+      def initialize
+        @throughput = Hash.new(0)
+        @latency = Hash.new { |hash, key| hash[key] = [] }
```
Comment (stamhankar999) [inline lib/metrics.rb]: Isn't this just an empty hash? Why the fancy initialization?

Comment (stamhankar999) [inline lib/metrics.rb]: Nvm. I get it now.

File: server.rb [hunk 3]
```diff
@@ -1,117 +1,348 @@
...
-            session.execute_async(
-              "SELECT username, firstname, lastname, password, email, created_date FROM videodb.users WHERE username = ?", arguments: [username]
-            ).then do |rows|
-              row = rows.first
-              row && row['username']
+  class GraphiteThread
+    def run(client, metrics, interval)
+      percentiles = [50, 60, 70, 80, 90, 95, 99, 99.9]
+
+      while true do
```
Comment (stamhankar999) [inline server.rb]: Use 'loop do' instead for unconditional infinite loops.

File: server.rb [hunk 4]
```diff
@@ -1,117 +1,348 @@
...
+              else
+                client.metrics({ name + '.latency.p' + percentile.to_s => latency[timestamp].percentile(percentile) },
+                               timestamp)
+              end
             end
+            client.metrics({ name + '.latency.avg' => latency[timestamp].mean }, timestamp)
+            client.metrics({ name + '.latency.max' => latency[timestamp].max }, timestamp)
+
+            # Throughput
+            client.metrics({ name + '.thoughput' => throughput[timestamp] }, timestamp)
```
Comment (stamhankar999) [inline server.rb]: typo: throughput

File: server.rb [hunk 5]
```diff
@@ -1,117 +1,348 @@
...
+    cluster.close
+    Thread.kill(thread)
+  end
+
+  TorqueBox::Web.run(
+    rack_app: App.run(session, OPTIONS[:experiment], OPTIONS[:statement], metrics),
+    host: '0.0.0.0',
+    port: 8080
+  ).run_from_cli
+end
```
Comment (stamhankar999) [inline server.rb]: Make sure this file ends in newline.

**Comment by stamhankar999** [comment]
Overall looks very good. Fix the minor issues and merge up!

File: Gemfile [hunk 6]
```diff
@@ -1,5 +1,7 @@
-ruby '1.9.3', engine: 'jruby', engine_version: '1.7.16'
+ruby '1.9.3', engine: 'jruby', engine_version: '1.7.24'
+ruby '2.2.3', engine: 'jruby', engine_version: '9.0.5.0'
 source "https://rubygems.org"
 
-gem 'cassandra-driver'
```
Comment (avalanche123) [inline Gemfile]: we don't depend on `cassandra-driver` anymore?

Comment (kishkaru) [inline Gemfile]: I'm running it directly from the source now, via `ruby -I path/to/driver/lib`. This way we can run the SUT against any driver version (released or not). Do you know of a better way to include an arbitrary version of the driver?

Comment (avalanche123) [inline Gemfile]: we can use Bundler's git support, e.g. `gem 'cassandra-driver', git: 'git://github.com/datastax/ruby-driver.git', ref: 'commit sha1'`, but what you have will also work

**Pull request merged**